java/code/src/com/redhat/rhn/frontend/action/user/CreateUserAction.java

```diff
@@ -167,7 +167,12 @@ public ActionForward execute(ActionMapping mapping,
         if (preferredLocale != null && preferredLocale.equals("default")) {
             preferredLocale = null;
         }
+        String preferredDocsLocale = form.getString("preferredDocsLocale");
+        if (preferredDocsLocale != null && preferredDocsLocale.equals("default")) {
+            preferredDocsLocale = null;
+        }
         user.setPreferredLocale(preferredLocale);
+        user.setPreferredDocsLocale(preferredDocsLocale);
```

Comment:
Wow, the logic here is a bit tricky:

`if A is not null and it is equals to something, then you set the initial A to null`

Why not  just comparing `default` string to the possible `null` one instead? Something like `"default".equals(preferredDocsLocale)` should not throw errors in case it is null and you can handle everything in one line then. Something like `user.setPreferredDocsLocale("default".equals(preferredDocsLocale) ? null : preferredDocsLocale)`, or even better using `Optionals` while fetching the `form.getString("preferredDocsLocale")`. I'm just suggesting though.

Reply:
I like your first option. It's really short. But IMO it is much harder to understand whats going on than using the `if(...)` syntax. Still going for your proposal (I like cryptic code after all :smile:)